homeassistant/components/sonos/media_player.py

```diff
@@ -395,11 +320,12 @@ def _timespan_secs(timespan: str | None) -> None | float:
     return sum(60 ** x[0] * int(x[1]) for x in enumerate(reversed(timespan.split(":"))))
 
 
-class SonosEntity(MediaPlayerEntity):
+class SonosMediaPlayerEntity(SonosEntity, MediaPlayerEntity):
     """Representation of a Sonos entity."""
 
-    def __init__(self, player: SoCo) -> None:
+    def __init__(self, player: SoCo, sonos_data: SonosData) -> None:
         """Initialize the Sonos entity."""
+        super().__init__(player, sonos_data)
         self._subscriptions: list[SubscriptionBase] = []
         self._poll_timer: Callable | None = None
         self._seen_timer: Callable | None = None
```

Comment:
This is duplicated functionality `elf._seen_timer` that can helpfully be moved to the base entity class

Reply:
This should really be cleaned up before merging.